zentral/contrib/santa/serializers.py

```diff
@@ -59,18 +62,264 @@ def update(self, instance, validated_data):
+        else:
+            # target does not exist on update
+            if not Target.objects.filter(type=target_type, identifier=target_identifier).exists():
+                raise serializers.ValidationError({"target": "does not exist!"})
+
+        # bundle target checks
+        if target_type == Target.BUNDLE:
+            try:
+                bundle = Bundle.objects.get(target__identifier=target_identifier)
+            except Bundle.DoesNotExist:
+                raise serializers.ValidationError({"target_type": f'Bundle for {target_identifier} does not exist.'})
+            else:
+                if not bundle.uploaded_at:
+                    raise serializers.ValidationError({"bundle": "This bundle has not been uploaded yet."})
+
+        # policy
+        try:
+            policy = int(data.get("policy"))
+        except (TypeError, ValueError):
+            pass
+        else:
+            # use custom message only on blocklist rules
+            if policy is not Rule.BLOCKLIST:
+                if data.get("custom_msg"):
+                    raise serializers.ValidationError({"custom_msg": "Can only be set on BLOCKLIST rules"})
+            if target_type is Target.BUNDLE:
+                if policy not in Rule.BUNDLE_POLICIES:
+                    raise serializers.ValidationError({"policy": f"Policy {policy} not allowed for bundles."})
+        
+        # add target_type, target_identifier for create
+        if not self.instance:
+            data["target_type"] = target_type
+            data["target_identifier"] = target_identifier
+
+        # remove target
+        data.pop("target")
+
+        # validate scope
+        self.validate_scope(data)
+        return data
+
+    def create(self, validated_data):
+        # create the target from target_type and target_identifier
+        target_type = validated_data.pop("target_type")
+        target_identifier = validated_data.pop("target_identifier")
+        target, _ = Target.objects.get_or_create(type=target_type, identifier=target_identifier)
+
+        rule = Rule.objects.create(
+            configuration=validated_data["configuration"],
+            target=target,
+            policy=validated_data["policy"],
+            custom_msg=validated_data.get("custom_msg", ""),
+            description=validated_data.get("description", ""),
+            serial_numbers=validated_data.get("serial_numbers") or [],
+            excluded_serial_numbers=validated_data.get("excluded_serial_numbers") or [],
+            primary_users=validated_data.get("primary_users") or [],
+            excluded_primary_users=validated_data.get("excluded_primary_users") or []
+        )
+        tags = validated_data.get("tags")
+        if tags:
+            rule.tags.set(tags)
+        excluded_tags = validated_data.get("excluded_tags")
+        if excluded_tags:
+            rule.excluded_tags.set(excluded_tags)
+        return rule
```

Comment:
If you want to add the events in this PR, you should also post an event here. see https://github.com/zentralopensource/zentral/blob/d301166954f7b754d90252791afb2bdca1ef4bcf/zentral/contrib/santa/views/setup.py#L294-L295